R/validate.R

```diff
@@ -552,3 +552,52 @@
   )
   return(NULL)
 }
+
+#' Validate each of the strata columns passed to baselinenowcast
+#'
+#' @inheritParams baselinenowcast.data.frame
+#' @keywords internal
+#' @returns NULL
+.validate_strata_cols <- function(strata_cols,
+                                  data) {
+  # Ensure nowcast_unit is not reference_date, report_date, count
+  conflicting_cols <- intersect(strata_cols, c(
+    "reference_date",
+    "report_date",
+    "count"
+  ))
+  if (length(conflicting_cols) > 0) {
+    cli_abort(
+      message = c(
+        "`strata_cols` cannot contain any of the required columns of: reference_date, report_date, count .", # nolint
+        "i" = "Found: {conflicting_cols} in `strata_cols`" # nolint
+      )
+    )
+  }
+
+  if (!all(strata_cols %in% colnames(data))) {
+    cli_abort(
+      message =
+        c("`strata_cols`, if specified, must be a column in `data`.",
+          "i" = "{strata_cols[!strata_cols %in% colnames(data)]} is not a column in `data`." # nolint
+        )
+    )
+  }
+  return(NULL)
+}
+
+#' Validate the date columns can be converted to Dates
+#'
+#' @param data Data.frame containing the date_col
+#' @param date_col Name of the column to be tested
+#' @keywords internal
+#' @returns NULL
+.validate_date_cols <- function(data, date_col) {
+  if (!inherits(data[[date_col]], "Date")) {
```

Comment:
you can't use is.Date?

Reply:
have removed `.validate_date_cols()` and swapped for `assert_date`